batchspawner/batchspawner.py

```diff
@@ -467,20 +481,20 @@ class BatchSpawnerRegexStates(BatchSpawnerBase):
         If this variable is set, the match object will be expanded using this string
         to obtain the notebook IP.
         See Python docs: re.match.expand""").tag(config=True)
+    state_unknown_re = Unicode('^$',
+        help="Regex that matches job_status if the resource manager is not answering").tag(config=True)
 
     def state_ispending(self):
         assert self.state_pending_re, "Misconfigured: define state_running_re"
-        if self.job_status and re.search(self.state_pending_re, self.job_status):
-            return True
-        else:
-            return False
+        return self.job_status and re.search(self.state_pending_re, self.job_status)
 
     def state_isrunning(self):
         assert self.state_running_re, "Misconfigured: define state_running_re"
-        if self.job_status and re.search(self.state_running_re, self.job_status):
-            return True
-        else:
-            return False
+        return self.job_status and re.search(self.state_running_re, self.job_status)
+
+    def state_isunknown(self):
+        assert self.state_unknown_re, "Misconfigured: define state_unknown_re"
+        return self.job_status and re.search(self.state_unknown_re, self.job_status)
```

Comment:
```suggestion
    def state_isunknown(self):
        if self.state_unknown_re:
            return self.job_status and re.search(self.state_unknown_re, self.job_status)
```
Changed so that if `state_unknown_re` is an empty string, it will always be false.

Does it need to `self.job_status and` here?  Let's see, this would only matter if it's an empty string.  In that case, I guess the regex hopefully wouldn't match anyway.  BUT- this saves us from an exception if self.job_status is `None`.  So I guess it's correct as-is. Anyone disagree?